exercises/concept/maze-maker/tests/Tests.elm

```diff
@@ -0,0 +1,187 @@
+            , fuzz
+                (Fuzz.fromGenerator (MazeMaker.branch MazeMaker.room))
+                "branch generates branches with rooms"
+              <|
+                \branch ->
+                    case branch of
+                        Branch branches ->
+                            List.all MazeMakerSupport.isRoom branches |> Expect.equal True
+
+                        _ ->
+                            Expect.fail "should only generate a Branch"
+            , fuzz
+                (Fuzz.fromGenerator (MazeMaker.branch MazeMaker.deadend))
+                "branch generates at least two branches"
+              <|
+                \branch -> MazeMakerSupport.branchNumber branch |> Expect.atLeast 2
+            , fuzz
+                (Fuzz.fromGenerator (MazeMaker.branch MazeMaker.deadend))
+                "branch generates at most four branches"
+              <|
+                \branch -> MazeMakerSupport.branchNumber branch |> Expect.atMost 4
+            , fuzzWith
+                { runs = 100
+                , distribution =
+                    Test.expectDistribution
+                        [ ( Test.Distribution.atLeast 30, "has 2 branches", (==) 2 )
+                        , ( Test.Distribution.atLeast 30, "has 3 branches", (==) 3 )
+                        , ( Test.Distribution.atLeast 30, "has 4 branches", (==) 4 )
+                        ]
+                }
+                (Fuzz.fromGenerator (MazeMaker.branch MazeMaker.deadend |> Random.map MazeMakerSupport.branchNumber))
+                "branch generates 2, 3, or 4 branches equally"
+              <|
+                \_ -> Expect.pass
+            ]
+        , describe "4"
+            [ fuzzWith
+                { runs = 100
+                , distribution =
+                    Test.expectDistribution
+                        [ ( Test.Distribution.atLeast 50, "is a deadend", (==) DeadEnd )
+                        , ( Test.Distribution.atLeast 10, "is a room", MazeMakerSupport.isRoom )
+                        , ( Test.Distribution.atLeast 20, "is a branch", MazeMakerSupport.isBranch )
+                        ]
+                }
+                (Fuzz.fromGenerator MazeMaker.maze)
+                "generates 60% or deadends, 15% of rooms and 25% of branches"
+              <|
+                \_ -> Expect.pass
+            , fuzzWith
+                { runs = 1000
+                , distribution =
+                    Test.expectDistribution
+                        [ ( Test.Distribution.moreThanZero, "depth 0", \m -> MazeMakerSupport.mazeDepth m == 0 )
+                        , ( Test.Distribution.moreThanZero, "depth 1", \m -> MazeMakerSupport.mazeDepth m == 1 )
+                        , ( Test.Distribution.moreThanZero, "depth 2", \m -> MazeMakerSupport.mazeDepth m == 2 )
+                        , ( Test.Distribution.moreThanZero, "depth 3", \m -> MazeMakerSupport.mazeDepth m == 3 )
```

Comment:
The probability to get a depth of 3 is the smallest, just about 3%. Doing 1000 runs gives a <99.99% chance of that happening.

Reply:
Maybe it is enough to test to depth 2 here?

Reply:
I mention the probability to get depth 3 in the next task, so I thought it would be nice to actually get it.

I thought 2 was a bit underwhelming, 1000 runs is really fast anyway, and a <0.01% chance of failure is reasonable.